plugins/org.python.pydev.core/src/org/python/pydev/core/docutils/ReversePyDocIterator.java

```diff
@@ -0,0 +1,232 @@
+/**
+ * Copyright (c) 2005-2012 by Appcelerator, Inc. All Rights Reserved.
+ * Licensed under the terms of the Eclipse Public License (EPL).
+ * Please see the license.txt included with this distribution for details.
+ * Any modifications to this file must keep this entire header intact.
+ */
+package org.python.pydev.core.docutils;
+
+import org.eclipse.jface.text.BadLocationException;
+import org.eclipse.jface.text.IDocument;
+import org.python.pydev.shared_core.string.StringUtils;
+
+public class ReversePyDocIterator implements IPyDocIterator {
+
+    private int offset;
+    private IDocument doc;
+
+    private boolean addNewLinesToRet = true;
+    private boolean returnNewLinesOnLiterals = false;
+    private boolean inLiteral = false;
+    private int literalStart = 0;
+    private boolean changeLiteralsForSpaces = false;
+    private int lastReturned = -1;
+    private boolean addComments = false;
+    private boolean considerAfterLiteralEnd = true;
+
+    public ReversePyDocIterator(IDocument doc, boolean addNewLinesToRet) {
+        this(doc, addNewLinesToRet, false, false);
+    }
+
+    public ReversePyDocIterator(IDocument doc, boolean addNewLinesToRet, boolean returnNewLinesOnLiterals,
```

Comment:
We don't need all those options for now. Please code it just for the specific use case you're using it (later on, if needed we can add more options).